programs/Makefile

```diff
@@ -1,23 +1,10 @@
-
-# To compile on SunOS: add "-lsocket -lnsl" to LDFLAGS
-
-CFLAGS	?= -O2
-WARNING_CFLAGS ?= -Wall -Wextra -Wformat=2 -Wno-format-nonliteral
-WARNING_CXXFLAGS ?= -Wall -Wextra -Wformat=2 -Wno-format-nonliteral
-LDFLAGS ?=
-
 MBEDTLS_TEST_PATH = ../tests
 # Support code used by test programs and test builds, excluding TLS-specific
 # code which is in the src/test_helpers subdirectory.
 MBEDTLS_TEST_OBJS = $(patsubst %.c,%.o,$(wildcard ${MBEDTLS_TEST_PATH}/src/*.c ${MBEDTLS_TEST_PATH}/src/drivers/*.c))
 
 LOCAL_CFLAGS = $(WARNING_CFLAGS) -I$(MBEDTLS_TEST_PATH)/include -I../include -D_FILE_OFFSET_BITS=64
-LOCAL_CXXFLAGS = $(WARNING_CXXFLAGS) -I../include -I../tests/include -D_FILE_OFFSET_BITS=64
-LOCAL_LDFLAGS = ${MBEDTLS_TEST_OBJS} 		\
-		-L../library 			\
-		-lmbedtls$(SHARED_SUFFIX)	\
-		-lmbedx509$(SHARED_SUFFIX)	\
-		-lmbedcrypto$(SHARED_SUFFIX)
+include ../scripts/common.make
```

Comment:
Also, in terms of readability, I think I'd find it a bit clearer for the include to be at the top of the file.

Reply:
In terms of readability, I would prefer to define the variables used by `common.make` before including `common.make`. That's not necessary for recursively expanded variables (i.e. lazy, `=`), only for simply expanded variables (i.e. strict, `:=`), but definition-before-use is simpler for humans and also avoids surprises if we switch a variable to be strict.

I'm going to keep the definition of `MBEDTLS_TEST_PATH` before the inclusion, but rework `MBEDTLS_TEST_OBJS` for better readability.